Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-usermenuview', function (Y) {
+    'use strict';
+    /**
+     * Provides the User Menu View class
+     *
+     * @module ez-usermenuview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_HIDDEN = 'is-user-menu-hidden',
+        CLASS_USER_PROFILE = 'ez-user-profile',
+        CLASS_USER_MENU = 'ez-user-menu',
+        SELECTOR_USER_PROFILE = '.' + CLASS_USER_PROFILE;
+
+    /**
+     * The User Menu view
+     *
+     * @namespace eZ
+     * @class UserMenuView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserMenuView = Y.Base.create('userMenuView', Y.View, [], {
+        _clickOutsideUserMenuHandler: null,
+
+        initializer: function () {
+            this.containerTemplate = '<ul class="ez-view-usermenuview"/>';
+
+            this.on('displayedChange', this._toggleUserMenu);
+            this.after('displayedChange', this._setClickOutsideUserMenuHandler);
+        },
+
+        /**
+         * Renders the user menu view
+         *
+         * @method render
+         * @return {eZ.UserMenuView} the view itself
+         */
+        render: function () {
+            this.get('container').addClass(CLASS_HIDDEN).addClass(CLASS_USER_MENU);
+
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Toggles the displayed value.
+         *
+         * @method toggleDisplayed
+         * @public
+         */
+        toggleDisplayed: function () {
+            this.set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _toggleUserMenu
+         * @protected
+         * @param event {Object} event facade
+         */
+        _toggleUserMenu: function (event) {
```

Comment:
to help readability, this kind of method is usually prefixed with `_ui` (so `_uiToggleUserMenu`) to reflect the fact it acts on the UI.